### PR TITLE
Run3-hcx369 Allow only valid ZDChits to enter the digitization phase

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Allow only valid ZDChits to enter the digitization phase

#### PR validation:

Use runTheMatrix test workflows

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

May be required to be back ported

### CODE REVIEW

**Comment by abdoulline** [comment]
Mat be it's worth checking the relevant wfs 180.1 and 181.1 ?  🤔

File: SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc [hunk 1]
```diff
@@ -420,7 +423,25 @@ void HcalDigitizer::accumulateCaloHits(edm::Handle<std::vector<PCaloHit>> const
...
     if (zdcgeo && theZDCDigitizer) {
-      theZDCDigitizer->add(*zdcHandle.product(), bunchCrossing, engine);
+      std::vector<PCaloHit> zdcHitsOrig = *zdcHandle.product();
+      std::vector<PCaloHit> zdcHits;
+      zdcHits.reserve(zdcHitsOrig.size());
+      // eliminate bad hits
+      for (unsigned int i = 0; i < zdcHitsOrig.size(); i++) {
+        DetId id(zdcHitsOrig[i].id());
+        HcalZDCDetId hid(id);
+        if (!ztopoP->valid(hid)) {
```
Comment (civanch) [inline SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc:433]: @bsunanda , in real run there will be many events. Is it correct to have unlimited printouts of wrong ZDC hits?

Comment (bsunanda) [reply SimCalorimetry/HcalSimProducers/src/HcalDigitizer.cc:433]: I shall protect the printout of good ones - but the wrong DetId is an error and should not happen in normal runs. We have similar error printouts for HCAL which do not produce large printouts

**Comment by bsunanda** [comment]
I have tested 180.1 with 5000 events before sending this PR. I shall do it for 181.1 for a similar number of events. I cannot do for larger statistics

**Comment by civanch** [comment]
+1

**Comment by mandrenguyen** [comment]
+1

**Comment by abdoulline** [comment]
> I have tested 180.1 with 5000 events before sending this PR. I shall do it for 181.1 for a similar number of events. I cannot do for larger statistics

Great!

**Comment by perrotta** [comment]
Thank you @bsunanda for the fix.
I see you prepared a backport for 14_0_X (which should help solving the problems we have for the production of the 2024 pp MC samples, if I understand it correctly). Please also submit a backport for CMSSSW_14_1_X, otherwise one release cycle will remain not updated between 14_0_X and 14_2_X

**Pull request merged**